### PR TITLE
NAS-105426 / 12.0 / Reflect unlock scenario in encryption summary for encrypted datasets

### PR DESCRIPTION
**Pull request by sonicaj**
This PR introduces changes so that we reflect which datasets will be unlocked in `pool.dataset.encryption_summary` based on the provided keys and current pool state.

### CODE REVIEW

File: src/middlewared/middlewared/plugins/pool.py [hunk 1]
```diff
@@ -2276,9 +2309,38 @@ def encryption_summary(self, job, id, options):
             results.append({
                 'name': ds_name, 'key_format': ZFSKeyFormat(data['key_format']['value']).value,
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/pool.py]: Can we format this according to PEP8 please?

File: src/middlewared/middlewared/plugins/pool.py [hunk 2]
```diff
@@ -2276,9 +2309,38 @@ def encryption_summary(self, job, id, options):
...
+        for ds in sorted(results, key=lambda d: d['name'].count('/')):
+            for i in range(1, ds['name'].count('/') + 1):
+                check = ds['name'].rsplit('/', i)[0]
+                if check in failed:
+                    failed.add(ds['name'])
+                    ds['unlock_error'] = f'Child cannot be unlocked when parent "{check}" is locked'
+
+            if ds['valid_key']:
+                if not ds['unlock_error']:
+                    ds['unlock_successful'] = not bool(ds['unlock_error'])
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/pool.py]: Can't it be anything but `True` here?

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/pool.py]: It will be `False` if `unlock_error` is set above in the loop

Comment (themylogin) [reply src/middlewared/middlewared/plugins/pool.py]: I mean, `not bool(ds['unlock_error'])` will always be `True` in this `if not ds['unlock_error']:` branch?

**Pull request merged**